Document package booking template, package step object, JSON tag (#132)
easolhq/easolhq.github.io: docs/reference/objects/package/step.md

---
layout: default
title: Package Step

> parent: Package

Reviewer: Could this sit at the top level so its clear this isn't accessed via the package drop?
```suggestion
parent: Objects
```

---

# Package Step
{: .d-inline-block }
object
{: .label .fs-1 }

Represents a creator-defined step along a [package]({% link docs/reference/objects/package.md %}) booking journey.

<br>

#### Attributes

## `package_step.id`
{: .d-inline-block }
string

> {: .label .fs-1 }
> The database id of the step.

Reviewer: nit: spacing
```suggestion
{: .label .fs-1 }

The database id of the step.
```

## `package_step.name`
{: .d-inline-block }
string

> {: .label .fs-1 }
> The name given to the step.

Reviewer: nit: spacing
```suggestion
{: .label .fs-1 }

The name given to the step.
```

## `package_step.tagline`
{: .d-inline-block }
string
{: .label .fs-1 }

The description given to the step.

> ## `package_step.product_groups`
> 
> {: .d-inline-block }

Reviewer: nit: spacing
```suggestion
## `package_step.product_groups`
{: .d-inline-block }
```

> Array of packaged products

Reviewer: Related to below comment, could this link to the relevant drop e.g.
```suggestion
Array of [Product]({% link docs/reference/objects/product/index.md %})s
```

{: .label .fs-1 }

> Each product containing variants added to the Package Step by the creator. Use `.variants` to get the variants added to the step for this product.

Reviewer: In the package template both `product` and `variant` are called on the `product_group` - should there be another page breaking down what is available via the `product_group` drop? Or is the `product_group` just a `product` drop and we could remove the `.product` when passing through the product in `item.product` below?
```
{% for item in package_step.product_groups %}
    ...
          item: item.product,
          variants: item.variants, 
    ...
{% endfor %}
```

Author: There should; it's a "step product drop" - adding :+1:

> ## `package_step.required_start_date`
> 
> {: .d-inline-block }

Reviewer: nit: spacing
```suggestion
## `package_step.required_start_date`
{: .d-inline-block }
```

Date
{: .label .fs-1 }

> Creators may specify that bookers must choose to book accommodation within a specific date range. This will contain the start date of that range.

Reviewer: The current wording suggests that the customer needs to book dates within a range rather than that the customers dates need to include the required dates, suggesting we reword slightly.
```suggestion
Creators may specify that customers must book accommodation dates which include a specific date range. This will return the start date of that range.
```

> ## `package_step.required_end_date`
> 
> {: .d-inline-block }

Reviewer: nit: spacing
```suggestion
## `package_step.required_end_date`
{: .d-inline-block }
```

Date
{: .label .fs-1 }

> Creators may specify that bookers must choose to book accommodation within a specific date range. This will contain the end date of that range.

Reviewer: The current wording suggests that the customer needs to book dates within a range rather than that the customers dates need to include the required dates, suggesting we reword slightly.
```suggestion
Creators may specify that customers must book accommodation dates which include a specific date range. This will return the end date of that range.
```

> ## `package_step.next_step_path`
> 
> {: .d-inline-block }

Reviewer: nit: spacing
```suggestion
## `package_step.next_step_path`
{: .d-inline-block }
```

URL
{: .label .fs-1 }

> URL pointing to the next page of the booking journey. Navigating to this URL is only possible if the booker has added enough/the correct elements to the cart to honor the requirements set by the creator for this step.

Reviewer: ```suggestion
URL pointing to the next page of the booking journey. Navigating to this URL is only possible if the customer has added enough/the correct elements to the cart to honour the requirements set by the creator for this step.
```

## `package_step.validation_errors`

{: .d-inline-block }
Array of strings
{: .label .fs-1 }

> If the booker has not added enough elements to the cart to honor the requirements set by the creator for this step, will contain an explanation of the fact.

Reviewer: ```suggestion
If the customer has not added enough elements to the cart to honour the requirements set by the creator for this step, will contain an explanation of the fact.
```
